### DISCUSSION TITLE
CONUS404: Incorrect CRS derived from netCDF file attributes

### DISCUSSION THREAD

**Discussion by ercas** [hytest-org/hytest#406] [Dataset Questions]
Hi, I have been having some issues trying to assign a CRS to the CONUS404 data. I have been following the approach from here to generate a PROJ4 string from the metadata embedded in the netCDF file: <https://fabienmaussion.info/2018/01/06/wrf-projection/> which has resulted in the following PROJ4 string:

`+proj=lcc +units=m +a=6370000.0 +b=6370000.0 +lat_1=30.0 +lat_2=50.0 +lat_0=39.100006103515625 +lon_0=-97.9000015258789 [+nadgrids=@null](mailto:+nadgrids=@null)`

This yields the following result, with the grid appearing very small and constrained to a small area near Minneapolis (Kansas):

![image](https://github.com/hytest-org/hytest/assets/13721056/d695aecd-7675-423c-9466-caf7f7ec0d6e)

How could I go about generating / finding the correct CRS for CONUS404? Any guidance would be appreciated. Thank you!

**Comment by amsnyder** [comment]
Hi @ercas - where are you accessing the CONUS404 files from?

**Reply by ercas** [reply]
Hi! I am accessing the files on the NCAR RDA Dataset Archive collection on Globus, specifically `wrf2d_d01_2020-01-01_00:00:00.nc`. It has these attributes in the metadata:

```
DX: 4000
DY: 4000
[...]
CEN_LAT: 39.1000061035156
CEN_LON: -97.8999938964844
TRUELAT1: 30
TRUELAT2: 50
MOAD_CEN_LAT: 39.1000061035156
STAND_LON: -97.9000015258789
POLE_LAT: 90
POLE_LON: 0
[...]
MAP_PROJ: 1
MAP_PROJ_CHAR: Lambert Conformal
```

**Comment by ercas** [comment]
Hi, I think I was able to fix the scale issue and now have the following:

`+proj=lcc +lat_0=39.1000061035156 +lon_0=-97.9000015258789 +lat_1=30 +lat_2=50 +x_0=0 +y_0=0 +R=6370000 +to_meter=4000 +no_defs`

![image](https://github.com/hytest-org/hytest/assets/13721056/0fb80405-b8ff-4325-8931-74e6b05bf872)

I'm unsure how to translate it to the correct false latitude and longitude though.

**Reply by kmsampson** [reply]
You seem to have the correct proj4 string, but incorrect origin (0,0). You just need the origin (upper left corner), which is `(-2734000,2030000)`. I truncated the sub-meter precision of the origin coordinate for clarity,

**Comment by amsnyder** [comment]
I haven't really worked with the RDA data - but we use the python package `cartopy` to pull out crs info from the CONUS404 zarr store. You can see an example of how we have done that here: https://hytest-org.github.io/hytest/dataset_access/conus404_spatial_aggregation.html#aggregate-conus404-to-feature-polygons. You might be able to set up something similar with the netcdfs from RDA.

In our example, we output `.to_wkt()`, but if you need a proj4 string, it looks like you can replace that with `.proj4_init` to get that.
